week7/app/src/main/java/com/isabelledionisius/myomelettapp/omeletts/MainActivity.kt

```diff
@@ -0,0 +1,62 @@
+package com.isabelledionisius.myomelettapp.omeletts
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.isabelledionisius.myomelettapp.R
+import com.isabelledionisius.myomelettapp.omelettdetails.DetailActivity
+import com.isabelledionisius.myomelettapp.omeletts.list.OmelettAdapter
+import com.isabelledionisius.myomelettapp.omeletts.model.Omelett
+
+/** Main activity of the list page
+ *
+ */
+
+abstract class MainActivity : AppCompatActivity(),
```

Comment:
Your app is crashing because you have Abstract here for the main activity. Otherwise it should be fine.

Reply:
What do you mean by saying "I haven't implemented the networking part yet". I have set up a folder "networking" in my app that contains the files that I tried to do similarly to the video course.  In this folder I have the class "Network Status Checker", the files "RemoteApi.kt and RetrofitBuilders.kt, and the interface "RemoteAPIService". It would be helpful to know what exactly is missing that I can specifically work on that. I am confident that I get it to work until Monday. I did get API to work in the past for iOS, so I am sure I get it work for Android ( even if I feel it is still more complicated than iOS).

Reply:
What I mean by saying you haven't implemented it is that you haven't connected any of the networking to the UI.

It's just sitting as unused and predefined now, but you haven't actually connected it, tested its data (at least I guess), and you're not displaying any of the data in the UI.